src/box/txn.h

```diff
@@ -333,6 +333,26 @@ struct autoinc_id_entry {
 	int64_t id;
 };
 
+/** Type of allocation, used to collect statistics. */
+enum tx_alloc_type {
+	/**
+	 * Allocation of txn_stmt.
+	 */
+	TX_ALLOC_STMT = 0,
+	/**
+	 * Allocation made with box_txn_alloc().
+	 */
+	TX_ALLOC_USER_DATA = 1,
+	/**
+	 * Allocation made by txns for internal purposes
+	 * (xrow_header, ev_timer, trigger, etc).
+	 */
+	TX_ALLOC_SYSTEM = 2,
+	TX_ALLOC_MAX = 3,
+};
+
+extern const char *tx_alloc_type_strs[];
```

Comment:
Nit: why not use `ENUM` and friends from src/trivia/util.h?

Reply:
AFAICS, `ENUM` builds strs from enum values' names, but I want them to be different.

Reply:
Ditto.